src/analyses/apron/relationAnalysis.apron.ml

```diff
@@ -252,11 +296,88 @@ struct
       if M.tracing then M.traceu "relation" "-> %a\n" D.pretty r;
       r
     )
+  (* replaces the pointer with our helper variable for tracking the pointer and  multiplies the second operand with the size of the target type *)
+  let replacePointerWithMapping expWithPointer sizeOfTyp = 
+    let rec replacePointer e = 
+      match e with 
+      | Lval (Var v, offset) ->
+        let castedPointer =  PointerMap.to_varinfo v in 
+        Lval (Var castedPointer, offset)
+      | BinOp (binop, e1, e2, typ) when binop = PlusPI || binop = IndexPI ->  (*pointer is always on the most left*)
+        let e2WithMult = BinOp (Mult, integer sizeOfTyp , CastE (!ptrdiffType ,e2), !ptrdiffType) in
+        BinOp (PlusA, replacePointer e1 , e2WithMult, typ) 
+      | BinOp (MinusPI, e1, e2, typ) -> 
+        let e2WithMult = BinOp (Mult, integer sizeOfTyp, CastE (!ptrdiffType ,e2), !ptrdiffType) in
+        BinOp (MinusA, replacePointer e1 , e2WithMult, typ) 
+      | e -> e
+    in
+    match expWithPointer with 
+    | CastE (t,e) -> replacePointer e 
+    | e -> replacePointer e
+
+
+  let pointerAssign ctx (v:varinfo) e = 
+    (* check if we assign to a global pointer add all possible addresses to escapedAllocSize to prevent them from being filtered *)
+    if not !allPointerEscaped && v.vglob then
+      (match ctx.ask (Queries.MayPointTo e) with
+       | a when not (Queries.AD.is_top a) ->  
+         Queries.AD.iter (function 
+             | Addr (v,o) -> 
+               escapedAllocSize := Set.add (AllocSize.to_varinfo v) !escapedAllocSize
+             | _ -> ()
+           )  a 
+       | _ -> allPointerEscaped := true); (* every allocated memory piece may escape *)
+
+    if GobConfig.get_bool "ana.apron.pointer_tracking" then (
+      if M.tracing then M.trace "OOB" "pointerAssign=%b" (GobConfig.get_bool "ana.apron.pointer_tracking");
+      match sizeOfTyp (Lval (Var v, NoOffset)) with 
+      | Some typSize -> 
+        let castedPointer = PointerMap.to_varinfo v in
+        let ctx = if not @@ RD.Tracked.varinfo_tracked v then 
+            let st = {ctx.local with rel = RD.add_vars ctx.local.rel [RV.local castedPointer]} in (* add temporary g#out *)
+            {ctx with local = st}
+          else 
+            ctx 
+        in
+        let replacedExp = replacePointerWithMapping e typSize in
+        if M.tracing then M.trace "malloc" "castedPointer=%a replacedExp %a\n" CilType.Varinfo.pretty castedPointer d_exp replacedExp;
+        assignVariable ctx (Var castedPointer,NoOffset) replacedExp 
+      | _ -> ctx.local
+    ) 
+    else ctx.local
+
+  (* Basic transfer functions. *)
+
+  let assign ctx (lv:lval) e =
+    match lv with
+    | (Var v, NoOffset) when isPointerType v.vtype 
```

Comment:
Yes, this may be generalized. As we discussed last week, I probably won't be able to implement this. Due to the fact that some issues are still open like, which ghost variable to assign this to if this evaluates to multiple variables and saving the offset of the evaluation.